controllers/volsync/vshandler.go

```diff
@@ -1253,10 +1253,14 @@ func (v *VSHandler) validateAndProtectSnapshot(
 
 	// Add ownerRef on snapshot pointing to the vrg - if/when the VRG gets cleaned up, then GC can cleanup the snap
 	// Add label to indicate that VolSync should not delete/cleanup this snapshot
-	err = util.NewResourceUpdater(volSnap).
-		AddOwner(v.owner, v.client.Scheme()).
-		AddLabel(VolSyncDoNotDeleteLabel, VolSyncDoNotDeleteLabelVal).
-		Update(v.ctx, v.client)
+	// Cross-namespace owner references are disallowed, so setting owner is skipped, when VRG is situated in admin
+	// namespace
```

Comment:
Good comment! helps to understand the meaning of v.vrgInAdminNamespace. Maybe later we can improve the function name to w.vrgInApplicationNamesapce() which make it more clear that the vrg can own resources in the application namespace.